libraries/scripting/jvm-host/src/kotlin/script/experimental/jvmhost/repl/legacyReplEvaluation.kt

```diff
@@ -78,6 +79,8 @@ class JvmReplEvaluator(
                 }
             }
             else ->
+                /// TODO: nothing is saved in history here, and this leads to compilation/evaluation history is out of sync
+                /// See ticket https://youtrack.jetbrains.com/issue/KT-36397
```

Comment:
It was the place where I identified the bug that was not too obvious for me when I discovered it, so it may be useful for the developer who will meet this bug again. But ok, I've deleted this commit. If you will consider it helpful, you may add it yourself then.